plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3select/S3SelectPushdown.java

```diff
@@ -106,9 +106,12 @@ private static boolean isInputFormatSupported(Properties schema)
     public static boolean isCompressionCodecSupported(InputFormat<?, ?> inputFormat, Path path)
     {
         if (inputFormat instanceof TextInputFormat) {
-            return getCompressionCodec((TextInputFormat) inputFormat, path)
+            var compressionCodec = getCompressionCodec((TextInputFormat) inputFormat, path);
+
+            // S3 Select supports the following formats: uncompressed, GZIP and BZIP2.
+            return compressionCodec.isEmpty() || compressionCodec
                     .map(codec -> (codec instanceof GzipCodec) || (codec instanceof BZip2Codec))
-                    .orElse(false); // TODO (https://github.com/trinodb/trino/issues/2475) fix S3 Select when file not compressed
+                    .orElse(false);
```

Comment:
`orElse` is useless since you already checked it's not empty

what about removing `compressionCodec.isEmpty() ||` and using `orElse(true)` here?

Reply:
Makes sense, fixed in the latest commit.